src/jit/lower.cpp

```diff
@@ -303,7 +303,7 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
 
             hiStore->gtOp.gtOp1 = hiRhs;
             hiStore->CopyCosts(tree);
-            hiStore->gtFlags |= GTF_VAR_DEF;
+            hiStore->gtFlags = tree->gtFlags;
```

Comment:
I think that the right fix is `hiStore->gtFlags |= (tree->gtFlags & (GTF_ALL_EFFECT|GTF_LIVENESS_MASK));`